### ISSUE TITLE
Log in problems with JoinDialog - again. Current Develop branch

### ISSUE THREAD

**Issue by Astagor** [havfo/multiparty-meeting#183]
Well I have run into a new problem, with the current 2020-04-07 18:30 DEVELOP branch
In the JoinDialog when I am on the screen to chose my name and  I click the login button, RoomClient.login() requests 
`/auth/login?peerId=ni8g86x6&roomId=null`

When I'm in the Lobby waiting, the URL is OK 
`/auth/login?peerId=ni8g86x6&roomId=test`

So on the JoinDialog on the screen with the choose name, RoomClient.login()  this_roomId is not set, as the join action has not yet been triggered. But the JoinDialog already knows the roomId.

In JoinDialog line 210 I have put this logging:
`	logger.error('ASTAGOR room=', room);
	logger.error('ASTAGOR roomId=', roomId);
	logger.error('ASTAGOR roomClient=', roomClient);` 

and then when opening the JoinDialog choose name I get:
`ASTAGOR room= +0ms {name: "", state: "new", locked: false, inLobby: false, signInRequired: false, …}

ASTAGOR roomId= +1ms test

 ASTAGOR roomClient= +0ms we {_signalingUrl: null, _closed: false, _produce: true, _forceTcp: false, _tracker: "wss://tracker.lab.vvc.niif.hu:443", …}`

as you can see roomId is set correctly by room.name is not set.

While in the Lobby I get:

`ASTAGOR room= +0ms {name: "", state: "new", locked: false, inLobby: false, signInRequired: false, …}

 ASTAGOR roomId= +1ms test

ASTAGOR roomClient= +0ms we {_signalingUrl: null, _closed: false, _produce: true, _forceTcp: false, _tracker: "wss://tracker.lab.vvc.niif.hu:443", …}`

I'm almost 100% sure this was working 1 day ago, but maybe to much testing

What I would recommend in JoinDialog to change:
`				<DialogTitle
					myPicture={myPicture}
					onLogin={() =>
					{
						loggedIn ? roomClient.logout() : roomClient.login();
					}}
					loggedIn={loggedIn}
				>`
to something like this
				<DialogTitle
					myPicture={myPicture}
					onLogin={() =>
					{
						loggedIn ? roomClient.logout() : roomClient.login(roomId);
					}}
					loggedIn={loggedIn}
				>

and in RoomClient  change from 
`	login()
	{
		const url = `/auth/login?peerId=${this._peerId}&roomId=${this._roomId}`;
		window.open(url, 'loginWindow');
	}
`

to 

	login(roomId)
	{
               var roomIdTemp = this._roomId;
               if (typeof(roomId) !== 'undefined' && this._roomId==null)  roomIdTemp = roomId;
		const url = `/auth/login?peerId=${this._peerId}&roomId=${roomIdTemp }`;

		window.open(url, 'loginWindow');
	}

So to summarize, RoomClient.login() should have a parameter that JoinDialog can pass, when RoomClient is still before RoomClient.join() in oder to generate a correct login URL with roomId.

**Comment by Astagor** [comment]
In fact this did not solve the whole problem. As now:
in JoinDialog on the the choose name screen I log in successfully, then I press join audio-video and get stopped in the Lobby, instead of being authorized and joining the room.

What am I doing wrong?

**Comment by Astagor** [comment]
In the end I have decided to deactivate the login on the JoinDialog choose name and only live login in Lobby.

I did it by changing
`{ window.config && window.config.loginEnabled && `
to
`{ window.config && window.config.loginEnabled && inLobby &&`

in JoinDialog DialogTitle. Of course passing the inLobby in props.

**Comment by Astagor** [comment]
Next thing I noticed. When in room and authenticated and with role admin, when I logout my role admin is not being removed. In Server.js in method app.get('/auth/logout', (req, res) =>
const { peerId } = req.session;
peerId is undefined.

I'm reverting to commit from 1 of April, before the introduction on the extended roles

**Comment by Astagor** [comment]
In the end I found what was causing the problem. I have switched in my nginx to proxy to 127.0.0.1 and used the new feature of setting the listeningHost in server/config.js.  Well, the session cookie is not set. This was the problem. Tomorrow I will have a look how to configure nginx that the session cookie is being set correctly.

This can be helpful for others who want to have nginx in front of multiparty meeting,  and have multiparty meeting listen only on 127.0.0.1.

**Comment by Astagor** [comment]
OK, I will leave this thread to the problem with the login button in JoinDialog.js.
I would introduce one more change to indicate when a user has logged in on "Choose Name" dialog. On a mobile he cannot see a difference, as he will not see the tooltip that it has changed to Logout. I would add style green to the icon if the user has logged in. If the user has a picture there is of course no problem.

`					<IconButton
						aria-label='Account'
						className={classes.loginButton}
						color='inherit'
						onClick={onLogin}
					>
						{            
	              myPicture ?
	              <Avatar src={myPicture} className={classes.largeAvatar} />
	              :
	                loggedIn ?
	                <AccountCircle className={[classes.largeIcon, classes.green]} />
	                :
	                <AccountCircle className={classes.largeIcon} />
						}
					</IconButton>`

**Comment by Astagor** [comment]
Thanks a lot it works perfectly. On last comment.
I have logged in, entered the room, icon is green. Now I press F5 and thus I leave the room and land on the JoinDialog on the "Choose Name" dialog. Then the icon and action shows not logged in, although I'm logged in. I had a look at the code and of course in this moment I'm before the peer creation, and thus the JoinDialog has no information, if I'm logged in or not. Maybe the the server should have a look at the session, if the user is authenticated, as Passport will deserialize the user from the session and put this information for the App.js.

**Comment by havfo** [comment]
At the moment we have no mechanism to check authentication status outside of a socket connection. We need an endpoint for checking that with a simple HTTP request in that case.

**Comment by Astagor** [comment]
OK. I think the users can live without it some time more. Not that crucial.
